src/core/gui/TextEditor.cpp

```diff
@@ -630,31 +607,10 @@ void TextEditor::findPos(GtkTextIter* iter, double xPos, double yPos) const {
 
 void TextEditor::contentsChanged(bool forceCreateUndoAction) {
     std::string currentText = getText()->getText();
-
-    // I know it's a little bit bulky, but ABS on subtracted size_t is a little bit unsafe
-    if (forceCreateUndoAction ||
-        ((lastText.length() >= currentText.length()) ? (lastText.length() - currentText.length()) :
-                                                       (currentText.length() - lastText.length())) > 100) {
-        if (!lastText.empty() && !this->undoActions.empty() &&
-            this->undoActions.front().get().getUndoText() != currentText) {
```

Comment:
The point is that TextEditor::undoActions stayed forever empty because of this test, so there was no undo-redo while editing text.

Now, fixing this at this point seems a bit pointless, as GtkTextBuffer implements an undo-redo mechanism in GTK 4.